R/add_hri_variables.R

```diff
@@ -0,0 +1,137 @@
+#' Flag non-Scottish residents
+#'
+#' @details The variable keep flag can be in the range c(0:4) where
+#' \itemize{
+#' \item{keep_flag = 0 when resident is Scottish}
+#' \item{keep_flag = 1 when resident is not Scottish}
+#' \item{keep_flag = 2 when the postcode is missing or a dummy, and the gpprac is missing}
+#' \item{keep_flag = 3 when the gpprac is not English and the postcode is missing}
+#' \item{keep_flag = 4 when the gpprac is not English and the postcode is a dummy}
+#' }
+#' The intention is to only keep the records where keep_flag = 0
+#'
+#' @param data A data frame, usually an SLF Individual File
+#' @param slf_pc_lookup_path The path to the Source postcode lookup, defaults
+#' to [get_slf_postcode_path()]
+#'
+#' @return A data frame with the variable 'keep_flag'
+flag_non_scottish_residents <- function(data,
+                                        slf_pc_lookup_path = get_slf_postcode_path()) {
+  check_variables_exist(data, c("postcode", "gpprac"))
+
+  # Make a lookup of postcode areas, which consist of the first characters
+  # of the postcode
+  pc_areas <- arrow::read_parquet(slf_pc_lookup_path) %>%
```

Comment:
Instead of passing the path, pass the actual data, we can set a default so if we don't pass anything it will still work as expected but it lets us set up the targets pipeline nicely.

```suggestion
                                        slf_pc_lookup = read_file(get_slf_postcode_path())) {
  check_variables_exist(data, c("postcode", "gpprac"))

  # Make a lookup of postcode areas, which consist of the first characters
  # of the postcode
  pc_areas <- slf_pc_lookup %>%
```